src/sage/modular/hecke/ambient_module.py

```diff
@@ -21,9 +21,8 @@
 from . import module
 from . import submodule
 
-import sage.modules.all
-
-import sage.rings.all
+from sage.modules.free_module import FreeModule, is_FreeModule
+from sage.rings.integer import Integer
 
```

Comment:
Couldn't we get rid of `is_FreeModule` on the way?

Reply:
Perhaps not in this PR...

Reply:
Right, separate PR